addons/binding/org.openhab.binding.onebusaway/src/main/java/org/openhab/binding/onebusaway/internal/handler/StopHandler.java

```diff
@@ -0,0 +1,273 @@
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        if (httpClient != null) {
+            httpClient.destroy();
+            httpClient = null;
+        }
+    }
+
+    /**
+     * Registers the listener to receive updates about arrival and departure times for its route.
+     *
+     * @param listener
+     * @return true if successful.
+     */
+    protected boolean registerRouteDataListener(RouteDataListener listener) {
+        if (listener == null) {
+            throw new IllegalArgumentException("It makes no sense to register a null listener!");
+        }
+        boolean added = routeDataListeners.add(listener);
+        if (added) {
+            String routeId = listener.getRouteId();
+            List<ObaStopArrivalResponse.ArrivalAndDeparture> copiedRouteData;
+            synchronized (routeData) {
+                copiedRouteData = Lists.newArrayList(routeData.get(routeId));
+            }
+            Collections.sort(copiedRouteData);
+            listener.onNewRouteData(routeDataLastUpdateMs, copiedRouteData);
+        }
+        return added;
+    }
+
+    /**
+     * Unregisters the listener so it no longer receives updates about arrival and departure times for its route.
+     *
+     * @param listener
+     * @return true if successful.
+     */
+    protected boolean unregisterRouteDataListener(RouteDataListener listener) {
+        return routeDataListeners.remove(listener);
+    }
+
+    /**
+     * Forced an update to be scheduled immediately, and reschedules our normal task to run again on our normal interval
+     * starting now.
+     */
+    protected void forceUpdate() {
+        if (pollingJob != null) {
+            pollingJob.cancel(true);
+            pollingJob = null;
+        }
+        scheduler.execute(pollingRunnable);
+        scheduleJob();
+    }
+
+    private ApiHandler getApiHandler() {
+        Bridge bridge = getBridge();
+        if (bridge == null || !(bridge.getHandler() instanceof ApiHandler)) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.CONFIGURATION_ERROR, "No API bridge available");
+            return null;
+        }
+        return (ApiHandler) bridge.getHandler();
+    }
+
+    private StopConfiguration loadAndCheckConfiguration() {
+        StopConfiguration config = getConfigAs(StopConfiguration.class);
+        if (config.getInterval() == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "interval is not set");
+            return null;
+        }
+        if (config.getStopId() == null) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.CONFIGURATION_ERROR, "stopId is not set");
+            return null;
+        }
+        return config;
+    }
+
+    private void scheduleJob() {
+        assert (pollingJob == null);
+        pollingJob = scheduler.scheduleWithFixedDelay(pollingRunnable, config.getInterval(), config.getInterval(),
```

Comment:
See above; this would be a waste of local resources and API requests (they have a limit too) since in order to know we are online, we have to make a data request anyway.  We also don't want to schedule our re-occurring jobs until we know we are online, as that would waste local resources running a job which is doomed to fail.